nix/build.nix

```diff
@@ -0,0 +1,153 @@
+# is the best I can get after multiple hours of trying. :).
+
+{
+  # Helpers
+  fd
+, lib
+, keep-directory-diff
+, nix-gitignore
+, stdenv
+, stdenvNoCC
+
+  # Actual derivation dependencies.
+, autoconf
+, automake
+, cacert
+, git
+, gnumake
+, llvmPackages
+, mtools
+, nasm
+, pkg-config
+}:
+
+let
+  # The actual current source but close to the repo state, i.e., no files from
+  # the bootstrap step.
+  currentRepoSrc = nix-gitignore.gitignoreSource [
+    # Prevent that the hash ov the sources change, when we change it in Nix.
+    "flake.nix"
+    "flake.lock"
+    "nix/"
+
+    # bootstrapped sources from ./bootstrap
+    "/build-aux/freestanding-toolchain"
+    "/freestanding-headers"
+    "/decompressor/cc-runtime"
+    "/limine-efi"
+    "/tinf"
+    "/common/flanterm"
+    "/stb"
+  ] ../.;
+
+  # Contains the sources downloaded by the Git submodule-like initialation done
+  # in ./bootstrap.
+  #
+  # ALWAYS update the hash when one of the network dependencies in ./bootstrap
+  # changes.
+  # bootstrappedSrcHash = lib.fakeHash;
+  bootstrappedSrcHash = "sha256-pMxjeYuNC8Yk2Qm46aCGO1UEwEiZ/RwsYypmmJQa0A4=";
+  # The full accumulated source tree to build Limine from.
+  bootstrappedSrc = stdenvNoCC.mkDerivation {
+    pname = "limine-src-bootstrapped";
+    version = "0.0.0";
+    src = currentRepoSrc;
+    nativeBuildInputs = [
+      cacert
+      fd
+      git
+      keep-directory-diff
+    ];
+    buildPhase = ''
+      runHook preBuild
+      # This derivation only contains downloaded sources.
+      bash ./bootstrap --skip-autoconf
+
+      keep-directory-diff ${currentRepoSrc} .
+
+      # When cloning, Git automatically creates hooks. Unfortunately, in a Nix
+      # environment / on a NixOS system, this includes Nix store paths.
+      # However, me must prevent to have any Nix store path inside the final
+      # directory, as otherwise we get the error
+      # "illegal path references in fixed-output derivation"! Further, we must
+      # remove all git artifacts (.git dirs) as they affect the hash of the
+      # derivation in a non-deterministic way.
+      fd -u --type=d "^.git$" --min-depth=2 . --exec rm -rf {}
+
+      # This should report nothing. Othewise, the Nix build will fail.
+      # grep -r /nix/store .
+
+      runHook postBuild
+    '';
+    dontPatchShebangs = true; # probalby unneeded
+    installPhase = ''
+      runHook preInstall
+      mkdir $out
+
+      cp -r . $out
+
+      runHook postInstall
+    '';
+    doCheck = false;
+    dontFixup = true;
+    # See "fixed output derivation".
+    outputHashAlgo = "sha256";
+    outputHashMode = "recursive";
+    outputHash = bootstrappedSrcHash;
+  };
+
+  # Common build dependencies apart from the compiler toolchain.
+  commonBuildDeps = [
+    autoconf
+    automake
+
+    gnumake
+    mtools
+    nasm
+    pkg-config # Checked for by ./configure but seems unused?!
+  ];
+in
+lib.makeOverridable stdenv.mkDerivation {
```

Comment:
What is this `makeOverridable` for?

Reply:
Background: By convention, all packages in Nix are overridable. `limine.override({ autoconf = myPatchedAutoconf })`

But I put it at the wrong place. Because `pkgs.callPackage` is effectively `lib.makeOverridable` plus dependency injection, it is not needed and the package is overridable.